client/keys/show.go

```diff
@@ -123,6 +131,26 @@ func runShowCmd(cmd *cobra.Command, args []string) (err error) {
 		printKeyInfo(info, bechKeyOut)
 	}
 
+	if isShowDevice {
+		if isShowPubKey {
+			return fmt.Errorf("the device flag (-d) can only be used for addresses not pubkeys")
+		}
+		if viper.GetString(FlagBechPrefix) != "acc" {
+			return fmt.Errorf("the device flag (-d) can only be used for accounts")
+		}
+		// Override and show in the device
+		if info.GetType() != keys.TypeLedger {
```

Comment:
You think this should be the first thing we check in this if block?

Reply:
well, it depends on what you want to inform first.. how to use the command line or the fact that the key is not the correct one.
I am happy to swap it.

Reply:
No big deal, just curious.